### PR TITLE
fix: correct testdata lookup for deeply nested types

### PR DESCRIPTION
**Pull request by grayside**
Supporting relative paths for testdata lookup was increasingly complicated. To simplify, this Pull Request requires `GENERATE_DATA_SOURCE` to be configured as an absolute path. To avoid the complexity and uniqueness of absolute paths, recommend setting it similar to `export GENERATE_DATA_SOURCE=$(realpath tmp/google-cloudevents)`

In the current behavior, nested types (e.g., the 4 types nested inside beyondcorp), fail to find the testdata root. This seems to lead to "sometimes failing", though I haven't identified why only sometimes. We do not currently have testdata for these types.

### CODE REVIEW

**Review by muncus — APPROVED** [review]
this seems like a good case for using arguments rather than environment variables 😆 

Please fix the redundant `Stat` before merging. for the realpath, i'll defer to your preference.

File: .github/workflows/generate.yml
Comment (grayside) [reply .github/workflows/generate.yml:56]: This step is needed because `env` can't be dynamic. I could concatenate variables here... could you tell me more about the preference for variable concat over realpath?

File: internal/testhelper/testhelper.go
Comment (grayside) [reply internal/testhelper/testhelper.go:29]: This variable is used for multiple things. It's only the validation testing that needs the path to be absolute, so I thought it would be helpful to folks finding success running the generator with a relative path without difficulty to understand there's something else to do.

Comment (grayside) [reply internal/testhelper/testhelper.go:29]: Follow-up discussion with @muncus , I answered the wrong question.

- `Stat` ensure the configuration of the tests is correct. If it fails we fail out of the tests.
- `ReadDir` loads testdata if there are any. If it fails we skip running tests and continue.

File: .github/workflows/generate.yml
Comment (grayside) [reply .github/workflows/generate.yml:56]: Filed follow-up https://github.com/googleapis/google-cloudevents-go/issues/150

**Pull request merged**
